app/screens/apps_form/apps_form_field.tsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import BoolSetting from '@components/widgets/settings/bool_setting';
+import TextSetting from '@components/widgets/settings/text_setting';
+import {ViewTypes} from '@constants/index';
+import {AppField, AppSelectOption} from '@mm-redux/types/apps';
+import {Theme} from '@mm-redux/types/preferences';
+import React from 'react';
+
+import AppFormSelector from './app_form_selector';
+
+const TEXT_DEFAULT_MAX_LENGTH = 150;
+const TEXTAREA_DEFAULT_MAX_LENGTH = 3000;
+
+export type Props = {
+    field: AppField;
+    name: string;
+    errorText?: React.ReactNode;
+    theme: Theme;
+
+    value: AppSelectOption | string | boolean | null;
```

Comment:
This can use the `AppFormValue` type specified in `mm-redux/types/apps.ts`, to which it looks like `boolean` needs to be added.